### DISCUSSION TITLE
SyncAPI - Offload Processing

### DISCUSSION THREAD

**Discussion by RoySalisbury** [aaronwmorris/indi-allsky#685] [General]
Can you explain the SyncAPI and setting up a "central" server?  Since I am running on Raspberry Pi's, the creating of timelapse and KeogramStarTrails does put a lot of load on the CPU.  

What I would like to do is offload all the "non real-time" processing off the local Pi, and onto a central server.  I can then host this on a much more powerful system so that it has the CPU it needs.  I also now have 2 all sky cameras setup (one monochrome and one color) that I can then access from a central system to view results.

I guess this is where the separation of tasks into docker containers would be very helpful, but I can also just disable the extra ffmpeg/keogram stuff on the Pi.

Roy

**Comment by aaronwmorris** [comment]
I am currently in the middle of creating some diagrams to show different use cases for indi-allsky.  Hopefully, I will have those uploaded soon.

As for the SyncAPI... it is not really meant to offload processing.  It is really meant to create an external web portal for people to be able to browse images and videos of multiple cameras.  The timelapses and keograms are still generated on the Raspi, just uploaded to a remote server for viewing.

Currently, you cannot start the capture service without a camera attached.  Theoretically, I could allow the indi-allsky service to start on the remote server with no camera and just accept manual jobs.  The timelapses and keograms have to be generated outside of the web environment (in an async process) because the job will run longer than 180 seconds, which is the limit for web requests.

There is an option to offload processing with indi-allsky.  indi-allsky is just an INDI client and you can run the indiserver solo on the Raspi and run the capture process and web server on a faster VM/container, connected over the network.  (This does not work with libcamera.)  Using this method, you could even use a Raspi Zero for the indiserver, which does not require much RAM.

**Reply by RoySalisbury** [reply]
> There is an option to offload processing with indi-allsky. indi-allsky is just an INDI client and you can run the indiserver solo on the Raspi and run the capture process and web server on a faster VM/container, connected over the network. (This does not work with libcamera.) Using this method, you could even use a Raspi Zero for the indiserver, which does not require much RAM.

This is actually something I would be intreated in trying.  Basically turning a PiZeroW2 into a psudo- ip camera (not exactly, but you get the idea).  I could then run multiple of these and use a much more powerful system to do all the heavy lifting.

**Comment by aaronwmorris** [comment]
Docker support is merged with #767
